ocaml/xapi/certs.ml

```diff
@@ -0,0 +1,414 @@
+(*
+ * Copyright (C) Citrix Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation; version 2.1 only. with the special
+ * exception on linking described in file LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details.
+ *)
+
+open Rresult
+
+let ( let* ) = R.bind
+
+let defer f = Fun.protect ~finally:f
+
+(* Pem provides parsing of *.pem files. The returned strings still need
+to be decoded *)
+module Pem : sig
```

Comment:
This module looks identical to ocaml/gencert/pem.ml, is there a reason you're not using that?

Reply:
Indeed, there is a lot of overlap. The gencert/pem.ml only reads the host *.pem but does not read CA certs from stunnel. I could change gencert/pem.ml to a wider interface and use that. The other reason is that I tried to develop this code independent of existing code.